lib/scala/semver/src/main/java/org/enso/semver/SemVer.java

```diff
@@ -0,0 +1,145 @@
+    private Optional<String> prefix;
+
+    private static final Pattern regex = Pattern.compile("([a-z]+-)?(\\d+)\\.(\\d+)\\.(\\d+)(-[a-z]+)?");
+
+    private SemVer(long major, long minor, long patch, Optional<String> preRelease, Optional<String> prefix) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.preRelease = preRelease;
+        this.prefix = prefix;
+    }
+
+    private SemVer(long major, long minor, long patch, Optional<String> preRelease) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.preRelease = preRelease;
+        this.prefix = Optional.empty();
+    }
+
+    private SemVer(long major, long minor, long patch) {
+        this.major = major;
+        this.minor = minor;
+        this.patch = patch;
+        this.preRelease = Optional.empty();
+        this.prefix = Optional.empty();
+    }
+
+    public static Try<SemVer> parse(String version) {
+        var match = regex.matcher(version);
+        if (match.find()) {
+            Optional<String> prefix = match.groupCount() > 0 && match.group(1) != null ? Optional.of(match.group(1)) : Optional.empty();
+            Optional<String> preRelease = match.groupCount() == 5 && match.group(5) != null ? Optional.of(match.group(5).substring(1)) : Optional.empty();
+            return Success.apply(new SemVer(
+                    Integer.valueOf(match.group(2)),
+                    Integer.valueOf(match.group(3)),
+                    Integer.valueOf(match.group(4)),
+                    preRelease,
+                    prefix));
+
+        } else {
+            return Failure.apply(new RuntimeException("Invalid version " + version));
+        }
+    }
+
+    public static SemVer of(int major) {
+        return new SemVer(major, 0, 0);
+    }
+
+    public static SemVer of(int major, int minor) {
+        return new SemVer(major, minor, 0);
+    }
+
+    public static SemVer of(int major, int minor, int patch) {
+        return new SemVer(major, minor, patch);
+    }
+
+    public static SemVer of(int major, int minor, int patch, String preReleaseVersion) {
+        return new SemVer(major, minor, patch, Optional.of(preReleaseVersion));
+    }
+
+    public boolean isGreaterThanOrEqual(SemVer v) {
+        return major > v.major || major == v.major && (
+                minor > v.minor || minor == v.minor && patch >= v.patch);
+    }
+
+    public boolean isGreaterThan(SemVer v) {
+        return major > v.major || major == v.major && (
+                minor > v.minor || minor == v.minor && patch > v.patch);
+    }
+
+    public boolean isLessThanOrEqual(SemVer v) {
+        return major < v.major || major == v.major && (
+                minor < v.minor || minor == v.minor && patch <= v.patch);
+    }
+
+    public boolean isLessThan(SemVer v) {
+        return major < v.major || major == v.major && (
+                minor < v.minor || minor == v.minor && patch < v.patch);
```

Comment:
This ordering ignores the pre release part, but the [SemVer spec specifies it takes part in the ordering](https://semver.org/#spec-item-11) see Point 3:
> When major, minor, and patch are equal, a pre-release version has lower precedence than a normal version:
>
> Example: 1.0.0-alpha < 1.0.0.

I think we should be consistent about this.

Reply:
I'm not sure but I thought nightly builds may have been relying on this.